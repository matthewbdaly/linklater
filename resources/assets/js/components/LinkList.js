// @flow
import React from 'react';
import LinkListItem from './LinkListItem';
import type { Link } from '../Types';

type Props = {
  links: Array<Link>,
  filter: string,
};

const LinkList = (props: Props) => {
  let itemNodes = props.links.filter((item) => {
    if (!props.filter) {
      return true;
    }
    return item.title.toLowerCase().includes(props.filter.toLowerCase());
  }).map((item) => {
    return <LinkListItem key={item.id} link={item.link} title={item.title} />;
  });
  return (
    <ul id="links" className="list-group">
      {itemNodes}
    </ul>
  );
};

export default LinkList;
