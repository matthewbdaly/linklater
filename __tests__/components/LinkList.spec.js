import React from 'react';
import {findDOMNode} from 'react-dom';
import LinkList from '../../resources/assets/js/components/LinkList';
import renderer from 'react-test-renderer';

describe('LinkList', () => {
  it('renders the link list', () => {
    const tree = renderer
      .create(<LinkList links={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
