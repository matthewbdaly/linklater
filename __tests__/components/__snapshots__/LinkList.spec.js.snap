// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList renders the link list 1`] = `
<ul
  className="list-group"
  id="links"
/>
`;
